### PR TITLE
Fix "No provider for InjectionToken mat-datepicker-scroll-strategy"

### PR DESCRIPTION
**Pull request by frontend-specialisten**
## Description

- Fix `No provider for InjectionToken mat-datepicker-scroll-strategy` (issue only seems to appear for the datepicker, which is why the provider is not added for the timepicker in this PR)
- Corrected naming of the `WattTimePickerModule`

### CODE REVIEW

**Comment by ManBearTM** [comment]
What happened with this commit?
https://github.com/Energinet-DataHub/greenforce-frontend/pull/681/commits/8e6887a217a76238c2a48204b3f36c44a713d4ba

File: libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/dh-market-participant-edit-actor.component.html [hunk 1]
```diff
@@ -14,6 +14,13 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 -->
+<watt-form-field>
+  <watt-label>Single date</watt-label>
+  <watt-timepicker [formControl]="exampleFormControlSingle"></watt-timepicker>
+  <watt-error *ngIf="exampleFormControlSingle?.errors?.required">
+    Date is required
+  </watt-error>
+</watt-form-field>
```
Comment (dzhavat) [inline libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/dh-market-participant-edit-actor.component.html:23]: nitpick: clean up :)

File: libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/dh-market-participant-edit-actor.component.ts [hunk 2]
```diff
@@ -138,6 +144,10 @@ export class DhMarketParticipantEditActorComponent {
     DhMarketParticipantActorMarketRolesComponentScam,
     DhMarketParticipantActorContactDataComponentScam,
     WattValidationMessageModule,
+    WattFormFieldModule,
+    WattDatepickerModule,
+    WattTimepickerModule,
+    ReactiveFormsModule,
```
Comment (dzhavat) [inline libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/dh-market-participant-edit-actor.component.ts:150]: nitpick: clean up :)

File: libs/dh/message-archive/feature-log-search/src/lib/dh-message-archive-log-search.component.html [hunk 3]
```diff
@@ -95,13 +95,15 @@ <h1>{{ transloco("title") }}</h1>
...
+        <!--
         <input
           #searchInputDateTimeFrom
           [(ngModel)]="this.searchCriteria.dateTimeFrom"
           wattInput
           type="text"
           name="dateTimeFrom"
-        />
+        />-->
+        <watt-datepicker></watt-datepicker>
```
Comment (dzhavat) [inline libs/dh/message-archive/feature-log-search/src/lib/dh-message-archive-log-search.component.html:106]: nitpick: clean up :)

File: libs/ui-watt/src/lib/components/input/datepicker/watt-datepicker.component.ts [hunk 4]
```diff
@@ -61,6 +62,7 @@ const danishLocaleCode = 'da';
     WattInputMaskService,
     WattRangeInputService,
     { provide: MatFormFieldControl, useExisting: WattDatepickerComponent },
+    MAT_DATEPICKER_SCROLL_STRATEGY_FACTORY_PROVIDER,
```
Comment (dzhavat) [inline libs/ui-watt/src/lib/components/input/datepicker/watt-datepicker.component.ts:65]: question: seems like this provider is already provided in the `MatDatepickerModule` (https://github.com/angular/components/blob/main/src/material/datepicker/datepicker-module.ts#L85). I wonder why is it needed again. Any idea?

**Review by dzhavat — CHANGES_REQUESTED** [review]

File: libs/dh/shared/data-access-msw/src/lib/mocks/market-participant/market-participant.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    }
+  ),
+  rest.get(
+    'https://localhost:5001/v1/MarketParticipant/organization/:organizationId/actor/:actorId',
+    (req, res, ctx) => {
+      return res(ctx.status(200), ctx.json(actorMock));
+      return res(
+        ctx.status(500),
+        ctx.json({ error: { message: 'Internal server error' } })
+      );
```
Comment (dzhavat) [inline libs/dh/shared/data-access-msw/src/lib/mocks/market-participant/market-participant.ts:68]: issue: SonarCloud complains about unreachable code. Maybe this should be commented out?

**Comment by frontend-specialisten** [comment]
> What happened with this commit? [8e6887a](https://github.com/Energinet-DataHub/greenforce-frontend/commit/8e6887a217a76238c2a48204b3f36c44a713d4ba)

Messed up commit has now been reverted.

**Review by dzhavat — DISMISSED** [review]

**Review by dzhavat — APPROVED** [review]

**Pull request merged**